modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaScreenContributorClassCheck.java

```diff
@@ -0,0 +1,159 @@
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeJavaScreenContributorClassCheck extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		String oldJavaClassContent = javaClass.getContent();
+
+		String newJavaClassContent = oldJavaClassContent;
+
+		Matcher matcher = _screenContributorPattern.matcher(
+			oldJavaClassContent);
+
+		if (matcher.find()) {
+			String javaTermContent = null;
+
+			List<String> oldVariables = new ArrayList<>();
+
+			List<String> oldMethods = new ArrayList<>();
+
+			for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+				if (childJavaTerm.isJavaVariable()) {
+					JavaVariable javaVariable = (JavaVariable)childJavaTerm;
+
+					javaTermContent = javaVariable.getContent();
+
+					oldVariables.add(javaTermContent);
+				}
+				else if (childJavaTerm.isJavaMethod()) {
+					JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+					javaTermContent = javaMethod.getContent();
+
+					oldMethods.add(javaTermContent);
+				}
+			}
+
+			newJavaClassContent = _formatContent(
+				oldVariables, oldJavaClassContent, oldMethods, matcher);
+		}
+
+		return StringUtil.replace(
+			content, oldJavaClassContent, newJavaClassContent);
+	}
+
+	protected String[] getNewImports() {
+		return new String[] {
+			"com.liferay.portal.settings.configuration.admin.display." +
+				"PortalSettingsConfigurationScreenFactory"
+		};
+	}
+
+	private static List<String> _formatMethodsLines(
+		List<String> overrideAnnotationMethods) {
+
+		List<String> methodsLines = new ArrayList<>();
+
+		for (String overrideAnnotationMethod : overrideAnnotationMethods) {
+			String[] methodLines = StringUtil.splitLines(
+				overrideAnnotationMethod);
+
+			for (int i = 0; i < methodLines.length; i++) {
+				methodLines[i] = StringUtil.removeFirst(
+					methodLines[i], StringPool.TAB);
+
+				if (StringUtil.equals(
+						methodLines[i], StringPool.CLOSE_CURLY_BRACE)) {
+
+					methodLines[i] = methodLines[i] + StringPool.NEW_LINE;
+				}
+
+				methodsLines.add(methodLines[i]);
+			}
+		}
+
+		return methodsLines;
+	}
+
+	private String _formatContent(
+		List<String> oldVariables, String oldJavaClassContent,
+		List<String> oldMethods, Matcher matcher) {
+
+		String clazzName = matcher.group(2);
+
+		if (clazzName.contains("ScreenContributor")) {
+			clazzName = StringUtil.removeSubstring(
+				clazzName, "ScreenContributor");
+		}
+		else {
+			clazzName = clazzName + "InnerClass";
+		}
+
+		String newMethods = StringBundler.concat(
+			"@Override\n\tprotected ConfigurationScreen ",
+			"getConfigurationScreen() {\n\t\treturn _portalSettings",
+			"ConfigurationScreenFactory.create(\n\t\t\tnew ", clazzName,
+			"());\n\t}\n\n\t@Reference\n\tprivate PortalSettings",
+			"ConfigurationScreenFactory _portalSettingsConfiguration",
+			"ScreenFactory;\n\n\t");
```

Comment:
@nicolas-sss, can't we use `StringBundler.append` instead as below?

Using it will bring more legibility